From 7859fe8d97ae45e9f64e272ba7502a171385e6e6 Mon Sep 17 00:00:00 2001
From: Sven Joachim <svenjoac@gmx.de>
Date: Fri, 8 Jul 2016 10:41:41 +0200
Subject: [PATCH] dh_installdocs: Fix regression wrt symlinked doc directory

If a package uses dh_installdocs' --link-doc option, install_dir could
fail with an error message from install(1), because the symlink might
be dangling.  Use ensure_docdir instead which was designed precisely
for this case.

Regression introduced in commit 71007f72da682dd9d7f932d81ca2ecd5d46a0e70.
---
 dh_installdocs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dh_installdocs b/dh_installdocs
index a42a98d..41e0883 100755
--- a/dh_installdocs
+++ b/dh_installdocs
@@ -227,7 +227,7 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 			$target_package = $dh{MAINPACKAGE};
 		}
 		my $target_dir = "$tmp/usr/share/doc/$target_package";
-		install_dir($target_dir);
+		ensure_docdir($target_package);
 		foreach my $doc (@docs) {
 			next if excludefile($doc);
 			next if -e $doc && ! -s $doc; # ignore empty files
-- 
2.8.1

